pkg/controller/elasticsearch/validation/autoscaling_validation.go

```diff
@@ -0,0 +1,281 @@
+				)
+			}
+			policyNames.Add(autoscalingSpec.Name)
+		}
+
+		// Validate the set of roles managed by this autoscaling policy.
+		if autoscalingSpec.Roles == nil {
+			errs = append(errs, field.Required(autoscalingSpecPath(i, "roles"), "roles field is mandatory"))
+		} else {
+			sort.Strings(autoscalingSpec.Roles)
+			if containsStringSlice(rolesSet, autoscalingSpec.Roles) {
+				//A set of roles must be unique across all the autoscaling policies.
+				errs = append(
+					errs,
+					field.Invalid(
+						autoscalingSpecPath(i, "name"),
+						strings.Join(autoscalingSpec.Roles, ","),
+						"roles set is duplicated"),
+				)
+			} else {
+				rolesSet = append(rolesSet, autoscalingSpec.Roles)
+			}
+		}
+
+		// Machine learning nodes must be in a dedicated tier.
+		if stringsutil.StringInSlice(esv1.MLRole, autoscalingSpec.Roles) && len(autoscalingSpec.Roles) > 1 {
+			errs = append(
+				errs,
+				field.Invalid(
+					autoscalingSpecPath(i, "name"), strings.Join(autoscalingSpec.Roles, ","),
+					"ML nodes must be in a dedicated autoscaling policy"),
+			)
+		}
+
+		if !(autoscalingSpec.NodeCount.Min >= 0) {
+			errs = append(
+				errs,
+				field.Invalid(
+					autoscalingSpecPath(i, "resources", "nodeCount", "min"),
+					autoscalingSpec.NodeCount.Min,
+					"min count must be equal or greater than 0",
+				),
+			)
+		}
+
+		if !(autoscalingSpec.NodeCount.Max > 0) {
+			errs = append(
+				errs,
+				field.Invalid(
+					autoscalingSpecPath(i, "resources", "nodeCount", "max"),
+					autoscalingSpec.NodeCount.Max,
+					"max count must be greater than 0"),
+			)
+		}
+
+		if !(autoscalingSpec.NodeCount.Max >= autoscalingSpec.NodeCount.Min) {
+			errs = append(
+				errs,
+				field.Invalid(autoscalingSpecPath(i, "resources", "nodeCount", "max"),
+					autoscalingSpec.NodeCount.Max,
+					"max node count must be an integer greater or equal than the min node count"),
+			)
+		}
+
+		// Validate CPU
+		errs = validateQuantities(errs, autoscalingSpec.CPU, i, "cpu", minCPU)
+
+		// Validate Memory
+		errs = validateQuantities(errs, autoscalingSpec.Memory, i, "memory", minMemory)
+
+		// Validate storage
+		errs = validateQuantities(errs, autoscalingSpec.Storage, i, "storage", minStorage)
+	}
+	return errs
+}
+
+// autoscalingSpecPath helps to compute the path used in validation error fields.
+func autoscalingSpecPath(index int, child string, moreChildren ...string) *field.Path {
+	return field.NewPath("metadata").
+		Child("annotations", `"`+esv1.ElasticsearchAutoscalingSpecAnnotationName+`"`).
+		Index(index).
+		Child(child, moreChildren...)
+}
+
+// validateQuantities ensures that a quantity range is valid.
+func validateQuantities(
+	errs field.ErrorList,
+	quantityRange *esv1.QuantityRange,
+	index int,
+	resource string,
+	minQuantity resource.Quantity,
+) field.ErrorList {
+	var quantityErrs field.ErrorList
+	if quantityRange == nil {
+		return errs
+	}
+
+	if !minQuantity.IsZero() && !(quantityRange.Min.Cmp(minQuantity) >= 0) {
```

Comment:
```suggestion
	if !minQuantity.IsZero() && quantityRange.Min.Cmp(minQuantity) < 0) {
```
to avoid the negation?